core/runtime/src/main/java/io/quarkus/runtime/shutdown/ShutdownRecorder.java

```diff
@@ -28,6 +28,9 @@ public void setListeners(List<ShutdownListener> listeners, boolean delayEnabled)
     }
 
     public static void runShutdown() {
+        if (shutdownListeners == null) { // when QUARKUS_INIT_AND_EXIT is used, ShutdownRecorder#setListeners has not been called
```

Comment:
Hmmm. In this case, we are sure that we don't want to run any of the shutdown listeners? None will have been registered and needed at this phase?

Reply:
That's how it's always has been.